wraper for ttl (#153)
V4Fire/Core: src/core/cache/README.md

# core/cache

## caches

This module provides the base interface for a [[Cache]] data structure: a simple in-memory key-value storage, which can be useful to organize cache data structures.
The submodules contain different implementations for that interface. The main module re-exports these implementations:

> * `AbstractCache` — an alias for [`core/cache/interface/Cache`](interface.ts);

Reviewer: Это не надо заменять

console.log(cache.keys().length); // 1
```

## Decorators

Also, the module provides a bunch of functions to decorate cache storage, like adding the `ttl` feature or persisting storing.

> ### core/cache/ttl

Reviewer: Я бы сделал папку /core/cache/decorators

Provides a decorator for any cache to add the feature of cache expiring.

```js
import SimpleCache from 'core/cache/simple';
import wrapCacheWithTTL from 'core/cache/ttl';

const

>   cache = wrapCacheWithTTL(new SimpleCache());

Reviewer: Нужен еще пример с дефолтным TTL и пример с коллизией свойств и разным TTL

console.log(cache.keys().length); // 1
```

## Decorators

Also, the module provides a bunch of functions to decorate cache storage, like adding the `ttl` feature or persisting storing.

> ### core/cache/ttl

Reviewer: Не поправил путь

Provides a decorator for any cache to add the feature of cache expiring.

```js
import SimpleCache from 'core/cache/simple';
import addTTL from 'core/cache/decorators/ttl';

> // Function addTTL accept cache and default ttl as second argument

Reviewer: ```suggestion
// The function `addTTL` accepts a cache object and a value for the default TTL as the second argument
```

const
  cache = addTTL(new SimpleCache(), 1000);

> // Method "add" accepts as the third optional parameter time until expiring the item to store in milliseconds.

Reviewer: ```suggestion
// The method "add" accepts as the third optional parameter time until expiring the item to store in milliseconds
```

cache.add('foo', 'bar1', {ttl: 500});

> // Additional method to clear the `ttl` descriptor from a cache item by the specified key

Reviewer: ```suggestion
// Additional method to remove the `ttl` descriptor from a cache item by the specified key
```
